src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -1000,603 +946,315 @@ private ImmutableArray<Symbol> CreateMembers()
 
-            return true;
-        }
-
-        public override int GetHashCode()
-        {
-            return _underlyingType.GetHashCode();
-        }
+                    if (tuple.Arity == RestPosition)
+                    {
+                        // Ensure all Rest extensions are tuples
+                        var tupleCompatibleType = EnsureRestExtensionsAreTuples(tuple);
+
+                        var extensionTupleElementTypes = tupleCompatibleType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics[RestPosition - 1].Type.TupleElementTypesWithAnnotations;
+                        var typesBuilder = ArrayBuilder<TypeWithAnnotations>.GetInstance(RestPosition - 1 + extensionTupleElementTypes.Length);
+                        typesBuilder.AddRange(tupleCompatibleType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics, RestPosition - 1);
+                        typesBuilder.AddRange(extensionTupleElementTypes);
+                        elementTypes = typesBuilder.ToImmutableAndFree();
+                    }
+                    else
+                    {
+                        elementTypes = tuple.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics;
+                    }
 
-        public override Accessibility DeclaredAccessibility
-        {
-            get
-            {
-                if (_underlyingType.IsErrorType())
-                {
-                    return Accessibility.Public;
+                    return elementTypes;
                 }
-                else
-                {
-                    return _underlyingType.DeclaredAccessibility;
-                }
-            }
-        }
-
-        public override bool IsStatic
-        {
-            get
-            {
-                return false;
-            }
-        }
-
-        public override bool IsAbstract
-        {
-            get
-            {
-                return false;
-            }
-        }
-
-        public override bool IsSealed
-        {
-            get
-            {
-                return true;
-            }
-        }
-
-        public override int Arity
-        {
-            get
-            {
-                return 0;
-            }
-        }
-
-        public override ImmutableArray<TypeParameterSymbol> TypeParameters
-        {
-            get
-            {
-                return ImmutableArray<TypeParameterSymbol>.Empty;
-            }
-        }
-
-        internal override ImmutableArray<TypeWithAnnotations> TypeArgumentsWithAnnotationsNoUseSiteDiagnostics
-        {
-            get
-            {
-                return ImmutableArray<TypeWithAnnotations>.Empty;
-            }
-        }
-
-        public override NamedTypeSymbol ConstructedFrom
-        {
-            get
-            {
-                return this;
             }
-        }
 
-        public override bool MightContainExtensionMethods
-        {
-            get
+            public ImmutableArray<FieldSymbol> TupleElements(NamedTypeSymbol tuple)
```

Comment:
>tuple [](start = 77, length = 5)

What if this argument doesn't match `TupleUnderlyingType`?

Reply:
This code is in `TupleUncommonData`, so we don't have a `TupleUnderlyingType`. We need to pass it in.

---
In reply to: [355732859](https://github.com/dotnet/roslyn/pull/39370#discussion_r355732859) [](ancestors = 355732859)